src/Microsoft.Data.Analysis/DataFrame.Join.cs

```diff
@@ -368,22 +403,62 @@ public DataFrame Merge(DataFrame other, string[] leftJoinColumns, string[] right
 
             DataFrame ret = new DataFrame();
 
-            //insert columns from left dataframe (this)
-            for (int i = 0; i < this.Columns.Count; i++)
+            PrimitiveDataFrameColumn<long> mapIndicesLeft = isLeftDataFrameRetained ? retainedRowIndices : supplementaryRowIndices;
+            PrimitiveDataFrameColumn<long> mapIndicesRight = isLeftDataFrameRetained ? supplementaryRowIndices : retainedRowIndices;
+
+            // Insert columns from left dataframe (this)
+            for (var i = 0; i < this.Columns.Count; i++)
```

Comment:
>var [](http://example.com/codeflow?start=17&length=3)

nit: why did you use var and removed the type here? using the type is better here.

Reply:
fixed